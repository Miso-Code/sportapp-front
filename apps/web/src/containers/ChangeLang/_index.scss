.config-lang {
	&-form {
		@apply mt-4;
	}

    &-submit{
        @apply mt-4;
    }
}
