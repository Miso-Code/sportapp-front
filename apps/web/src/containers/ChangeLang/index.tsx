import { yupResolver } from '@hookform/resolvers/yup'
import { Box, Typography } from '@mui/material'
import { useForm } from 'react-hook-form'
import { useTranslation } from 'react-i18next'
import schema, { FormData } from './utils/schema'
import LoadingButton from '@mui/lab/LoadingButton'
import RadioButtonController from '@/components/Inputs/RadioButtonController'
import './_index.scss'
import { useLangStore } from '@sportapp/stores'

export default function ChangeLang() {
	const { t } = useTranslation()
	const { changeLang } = useLangStore()
	const { lang } = useLangStore()
	const { handleSubmit, control } = useForm<FormData>({
		resolver: yupResolver(schema),
		values: {
			lang
		},
		mode: 'onChange'
	})

	const onSubmit = (data: FormData) => {
		changeLang(data.lang)
	}
	return (
		<Box
			sx={{
				p: 4
			}}>
			<Typography className='config-lang-title' variant='h3'>
				{t('config.menu.language')}
			</Typography>

			<form
				className='config-lang-form'
				onSubmit={handleSubmit(onSubmit)}>
				<RadioButtonController
					control={control}
					fullWidth
					label={t('config.language.select')}
					name='lang'
					options={[
						{ label: t('config.language.en'), value: 'en' },
						{ label: t('config.language.es'), value: 'es' }
					]}
				/>

				<LoadingButton
					fullWidth
					className='config-lang-submit'
					size='large'
					type='submit'
					variant='contained'>
					{t('config.language.button')}
				</LoadingButton>
			</form>
		</Box>
	)
}
