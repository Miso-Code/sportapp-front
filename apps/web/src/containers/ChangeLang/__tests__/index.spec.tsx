import ChangeLang from '..'
import { render, RenderResult } from '@testing-library/react'

describe('ChangeLangContainer', () => {
	let renderResult: RenderResult

	beforeEach(() => {
		renderResult = render(<ChangeLang />)
	})

	it('should render the page', () => {
		expect(renderResult.container).toMatchSnapshot()
	})
})
