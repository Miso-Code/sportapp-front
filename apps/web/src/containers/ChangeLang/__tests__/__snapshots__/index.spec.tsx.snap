// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeLangContainer should render the page 1`] = `
<div>
  <div
    class="MuiBox-root css-1wcaknn"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 config-lang-title css-gepadz-MuiTypography-root"
    >
      config.menu.language
    </h3>
    <form
      class="config-lang-form"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
          id="lang-label"
        >
          config.language.select
        </label>
        <div
          aria-labelledby="lang-label"
          class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="lang"
                type="radio"
                value="en"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              config.language.en
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="lang"
                type="radio"
                value="es"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              config.language.es
            </span>
          </label>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
        />
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth config-lang-submit css-sppx0e-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
        id=":r1:"
        tabindex="0"
        type="submit"
      >
        config.language.button
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</div>
`;
